### DISCUSSION TITLE
Support for triggering refreshes of BI datasets (PowerBI, Tableau etc) as part of a pipeline.

### DISCUSSION THREAD

**Discussion by acarter24** [dagster-io/dagster#15636] [Ideas]
I'm using dagster to orchestrate the creation of datasets, but the 'final hop' of pulling the data into PowerBI is being done on a scheduled job, which feels a bit unsatisfying!

It would be great to be able to define a PowerBI dataset by name as an asset (with one or more upstream tables), and see the full lineage. That way a single click of 'materialize' can run the full pipeline from source API to bi dataset, and I can use freshness definitions to trigger a PowerBI refresh more often if volume demands it.

I've searched issues and the slack channel, but couldn't find anything relating to this, either for PowerBI or Tableau. Has anyone built such an integration?

**Comment by HuyNguyenNeyuData** [comment]
Have you tried using Power Bi API?
https://learn.microsoft.com/en-us/rest/api/power-bi/

**Comment by Elliot2718** [comment]
@acarter24 we've built a PowerBI integration for dagster internally, i.e. basically just assets that call the PowerBI API to refresh datasets or data flows. I'd be happy to show you how we did it if you're interested.

**Reply by leetyron777** [reply]
Hi @Elliot2718 . Am interested in the solution you have as mentioned above. are you able to share a code / source to this question?

**Reply by acarter24** [reply]
Likewise would be great if you could share here.

**Reply by acarter24** [reply]
@Elliot2718 any chance you could share some examples for this?

**Reply by C-Roensholt** [reply]
@Elliot2718 I would also love to see an example if this :)

**Reply by mgtaylor33** [reply]
I would also love to see an example of this please.

**Reply by geo909** [reply]
@Elliot2718 +1 one to the requests here. Can you kindly share your work?
